misk/src/test/kotlin/misk/digester/SlidingWindowDigestTest.kt

```diff
@@ -0,0 +1,232 @@
+        newWindowDigest(windowsT3_5[0], listOf(10.0, 200.0)),
+        newWindowDigest(windowsT3_5[1], listOf(200.0)),
+        newWindowDigest(windowsT3_5[2], listOf(200.0)
+        )))
+  }
+
+  @Test
+  fun testSlidingWindowDigestGC() {
+    val digest = newSlidingWindowDigestTest()
+    digest.observe(10.0)
+    // Move just past the threshold for collecting the last window
+    digest.utcNowClock = Clock.fixed(windows(digest)[2].end.plusMinutes(1).plusNanos(1).toInstant(),
+        ZoneId.of("UTC"))
+    digest.observe(20.0)
+    val windows = windows(digest)
+    expectWindowDigests(digest.windows, listOf(
+        newWindowDigest(windows[0], listOf(20.0)),
+        newWindowDigest(windows[1], listOf(20.0)),
+        newWindowDigest(windows[2], listOf(20.0))
+    ))
+  }
+
+  fun setClock(t: ZonedDateTime, digest: SlidingWindowDigest) {
+    require(!t.toInstant().isBefore(digest.utcNowClock.instant())) {
+      "Cannot go back in time"
+    }
+
+    digest.utcNowClock = Clock.fixed(t.toInstant(), ZoneId.of("UTC"))
+  }
+
+  fun windows(slidingWindow: SlidingWindowDigest): List<Window> {
+    return slidingWindow.windower.windowsContaining(
+        ZonedDateTime.ofInstant(slidingWindow.utcNowClock.instant(), ZoneId.of("UTC")))
+  }
+
+  fun advanceWindows(n: Int, digest: SlidingWindowDigest): List<Window> {
+    repeat(n) {
+      setClock(windows(digest)[0].end, digest)
+    }
+    return windows(digest)
+  }
+
+  fun newWindowDigest(window: Window, values: List<Double>): WindowDigest {
+    return WindowDigest(
+        window,
+        FakeDigest(values)
+    )
+  }
+
+  fun newSlidingWindowDigestTest(): SlidingWindowDigest {
+    return SlidingWindowDigest(
+        baseClock,
+        Windower(10, 3),
+        fun() = FakeDigest()
+    )
+  }
+
+  fun expectWindowDigests(actual: List<WindowDigest>, expected: List<WindowDigest>) {
+    Assertions.assertThat(expected.count()).isEqualTo(actual.count())
+    for (i in 0 until actual.count()) {
+      if (i >= expected.count()) {
+        break
+      }
+      //Compare window of each WindowDigest
+      Assertions.assertThat(actual[i].window).isEqualTo(expected[i].window)
+      //Compare all values added within TDigest of each WindowDigest
+      Assertions.assertThat((actual[i].Digest as FakeDigest).addedValues)
+          .isEqualTo((expected[i].Digest as FakeDigest).addedValues)
+    }
+  }
+
+  fun expectQuantiles(
+    digest: SlidingWindowDigest,
+    count: Long,
+    sum: Double,
+    quantileVals: Map<Double, Double>
+  ) {
+    val snapshot = digest.snapshot(quantileVals.keys.toList()) //should this be keys or values?
+    Assertions.assertThat(snapshot.count).isEqualTo(count)
+    assertEqualish(sum, snapshot.sum)
+    quantileVals.keys.forEachIndexed { i, q ->
+      assertEqualish(quantileVals[q], snapshot.quantileVals[i])
+    }
+  }
+
+  fun assertEqualish(a: Double?, b: Double?) {
+    if (a == Double.NaN) {
+      Assertions.assertThat(b).isEqualTo(Double.NaN)
```

Comment:
Static import so that the full line is `assertThat(b).isEqualTo(Double.NaN)`. Also, `assertThat(b).isNan()` since NaN != NaN.